# HG changeset patch
# User Philip Nienhuis <prnienhuis@@users.sf.net>
# Date 1312900930 25200
#      Tue Aug 09 07:42:10 2011 -0700
# Node ID 3509cf60d1f690444f418fa30ab590376dd35668
# Parent  9f2679a1436686171d1facd918acf83d0347c31d
strread.m: %Ns conversion format now honors user specified width (Bug #33950).

* strread.m: Fix typo in code which fixed width at 3, regardless of user input.

diff --git a/scripts/io/strread.m b/scripts/io/strread.m
--- a/scripts/io/strread.m
+++ b/scripts/io/strread.m
@@ -466,17 +466,17 @@ function varargout = strread (str, forma
             if (ii < numel (fmt_words))
               iwrdl = length (words{iwrd});
             endif
           endif
 
         endif
       endfor
       ## Done
-      words_period = iwrd - 1;
+      words_period = max (iwrd - 1, 1);
       num_lines = ceil (num_words / words_period);
 
       ## 2. Pad words array so that it can be reshaped
       tmp_lines = ceil (num_words / words_period);
       num_words_padded = tmp_lines * words_period - num_words;
       if (num_words_padded)
         words = [words'; cell(num_words_padded, 1)]; 
       endif
@@ -628,17 +628,17 @@ function varargout = strread (str, forma
                 data = int32 (data);
               end
               varargout{k} = data.';
               k++;
             case "s"
               if (pad_out)
                 data(end+1:num_lines) = {""}
               endif
-              varargout{k} = strtrunc (data, 3)';
+              varargout{k} = strtrunc (data, swidth)';
               k++;
             otherwise
           endswitch
         case {"%*", "%*s"}
           ## skip the word
         otherwise
           ## Ensure descriptive content is consistent.
           ## Test made a bit lax to accomodate for incomplete last lines
